src/util/bip32.rs

```diff
@@ -612,7 +627,7 @@ impl ExtendedPrivKey {
 
 impl ExtendedPubKey {
     /// Derives a public key from a private key
-    pub fn from_private<C: secp256k1::Signing>(secp: &Secp256k1<C>, sk: &ExtendedPrivKey) -> ExtendedPubKey {
+    pub fn from_priv<C: secp256k1::Signing>(secp: &Secp256k1<C>, sk: &ExtendedPrivKey) -> ExtendedPubKey {
```

Comment:
I am not a big fan of this change. But I can live with this. This basically brings a compile error downstream that is difficult to solve. I think we should at least add this in release notes. 

The user at least has some guidance from the compiler when we change the function signature while keeping the name the same. In this case, it would be just say function not found.